homework_4/bet_acceptance_api/cmd/bootstrap/api.go

```diff
@@ -0,0 +1,43 @@
+package bootstrap
+
+import (
+	"code-cadets-2021/homework_4/bet_acceptance_api/cmd/config"
+	"code-cadets-2021/homework_4/bet_acceptance_api/internal/api"
+	"code-cadets-2021/homework_4/bet_acceptance_api/internal/api/controllers"
+	"code-cadets-2021/homework_4/bet_acceptance_api/internal/api/controllers/validators"
+	"code-cadets-2021/homework_4/bet_acceptance_api/internal/domain/services"
+	"code-cadets-2021/homework_4/bet_acceptance_api/internal/infrastructure/rabbitmq"
+	"github.com/streadway/amqp"
+)
+
+func newBetAcceptanceValidator(coeffUpperBound, paymentLowerBound, paymentUpperBound float64) *validators.BetAcceptanceValidator {
+	return validators.NewBetAcceptanceValidator(coeffUpperBound, paymentLowerBound, paymentUpperBound)
+}
+
+func newBetReceivedPublisher(publisher rabbitmq.QueuePublisher) *rabbitmq.BetReceivedPublisher {
+	return rabbitmq.NewEventUpdatePublisher(
+		config.Cfg.Rabbit.PublisherExchange,
+		config.Cfg.Rabbit.PublisherBetReceivedQueue,
+		config.Cfg.Rabbit.PublisherMandatory,
+		config.Cfg.Rabbit.PublisherImmediate,
+		publisher,
+	)
+}
+
+func newBetAcceptanceService(publisher services.BetReceivedPublisher) *services.BetAcceptanceService {
+	return services.NewBetAcceptanceService(publisher)
+}
+
+func newController(betAcceptanceValidator controllers.BetAcceptanceValidator, betAcceptanceService controllers.BetAcceptanceService) *controllers.Controller {
+	return controllers.NewController(betAcceptanceValidator, betAcceptanceService)
+}
+
+// Api bootstraps the http server.
+func Api(rabbitMqChannel *amqp.Channel) *api.WebServer {
+	betAcceptanceValidator := newBetAcceptanceValidator(config.Cfg.BetValidator.CoefficientUpperBound, config.Cfg.BetValidator.PaymentLowerBound, config.Cfg.BetValidator.PaymentUpperBound)
```

Comment:
Just a tiny nitpick, you usually pass to this function stuff which you can't access in the function itself. If you take a look at the `newBetReceivedPublisher` there you only inject  a `QueuePublisher` because you can access the `config.Cfg` values from within the function. It just makes the bootstrapping a bit clearer, but like I said, it's a small nitpick you don't have to fix.

Reply:
Fixed it!